plugins/account_history/plugin.cpp

```diff
@@ -135,242 +157,290 @@ if( options.count(name) ) { \
+            impacted.insert(make_pair(op.to, operation_direction_type::receiver));
+            impacted.insert(make_pair(op.agent, operation_direction_type::receiver));
         }
 
         void operator()(const escrow_dispute_operation& op) {
-            impacted.insert(op.from);
-            impacted.insert(op.to);
-            impacted.insert(op.agent);
+            impacted.insert(make_pair(op.from, operation_direction_type::sender));
+            impacted.insert(make_pair(op.to, operation_direction_type::receiver));
+            impacted.insert(make_pair(op.agent, operation_direction_type::receiver));
         }
 
         void operator()(const escrow_release_operation& op) {
-            impacted.insert(op.from);
-            impacted.insert(op.to);
-            impacted.insert(op.agent);
+            impacted.insert(make_pair(op.from, operation_direction_type::sender));
+            impacted.insert(make_pair(op.to, operation_direction_type::receiver));
+            impacted.insert(make_pair(op.agent, operation_direction_type::receiver));
         }
 
         void operator()(const transfer_to_savings_operation& op) {
-            impacted.insert(op.from);
-            impacted.insert(op.to);
+            if (op.from == op.to) {
+                impacted.insert(make_pair(op.from, operation_direction_type::any));
+                impacted.insert(make_pair(op.to, operation_direction_type::any));
+            } else {
+                impacted.insert(make_pair(op.from, operation_direction_type::sender));
+                impacted.insert(make_pair(op.to, operation_direction_type::receiver));
+            }
         }
 
         void operator()(const transfer_from_savings_operation& op) {
-            impacted.insert(op.from);
-            impacted.insert(op.to);
+            if (op.from == op.to) {
+                impacted.insert(make_pair(op.from, operation_direction_type::any));
+                impacted.insert(make_pair(op.to, operation_direction_type::any));
+            } else {
+                impacted.insert(make_pair(op.from, operation_direction_type::sender));
+                impacted.insert(make_pair(op.to, operation_direction_type::receiver));
+            }
         }
 
         void operator()(const cancel_transfer_from_savings_operation& op) {
-            impacted.insert(op.from);
+            impacted.insert(make_pair(op.from, operation_direction_type::any));
         }
 
         void operator()(const decline_voting_rights_operation& op) {
-            impacted.insert(op.account);
+            impacted.insert(make_pair(op.account, operation_direction_type::any));
         }
 
         void operator()(const comment_benefactor_reward_operation& op) {
-            impacted.insert(op.benefactor);
-            impacted.insert(op.author);
+            impacted.insert(make_pair(op.benefactor, operation_direction_type::receiver));
+            impacted.insert(make_pair(op.author, operation_direction_type::sender));
```

Comment:
Looks like there is no restriction on `op.benefactor == op.author`, @afalaleev, is it?
If so, add `any` case.

Reply:
It is really funny, but the author can be really a benefactor.
Ok, lets add check on `op.benefactor == op.author`.